### PR TITLE
Replace Hard-Coded Sensible Capacity Curves with ADP/BF

### PR DESCRIPTION
**Pull request by joseph-robertson**
## Pull Request Description

[improves the design airflow/design SHR calculation in apply_hvac_equipment_adjustments within hvac_sizing.rb for central-ducted air systems used for cooling]

## Checklist

PR Author: Check these when they're done. Not all may apply. ~~strikethrough~~ and check any that do not apply. 

PR Reviewer: Verify each has been completed.

- [x] ~~Schematron validator (`EPvalidator.xml`) has been updated~~
- [x] ~~Sample files have been added/updated (`openstudio tasks.rb update_hpxmls`)~~
- [x] ~~Tests have been added/updated  (`HPXMLtoOpenStudio/tests/test_hvac_sizing.rb` and/or `workflow/tests/test*.rb`)~~
- [x] Documentation has been updated (no change necessary)
- [x] ~~Changelog has been updated~~
- [x] `openstudio tasks.rb update_measures` has been run
- [x] No unexpected changes to simulation results of sample files. TODO: ~~merge and test w/ diverse set of buildings/climates in ResStock~~ house 237.xml (Florida) edge case was investigated, and the ADP/BF approach for this case produces more reasonable equipment capacities (see attached excel, ppt).

[ADP_BF_ResStock Testing Results.pptx](https://github.com/user-attachments/files/17078608/ADP_BF_ResStock.Testing.Results.pptx)

[testing_ResStock_datapoint_237.xlsx](https://github.com/user-attachments/files/17078489/testing_ResStock_datapoint_237.xlsx)

### CODE REVIEW

File: HPXMLtoOpenStudio/resources/hvac.rb [hunk 1]
```diff
@@ -3333,7 +3337,7 @@ def self.set_cool_rated_shrs_gross(cooling_system)
       clg_ap.cool_rated_shrs_gross = [cooling_system.cooling_shr] # We don't model the fan separately, so set gross == net
     else
       # rated shr gross and fan speed ratios
-      dB_rated = 80.0 # F
+      dB_rated = HVAC::AirSourceCoolRatedIDB
```
Comment (shorowit) [inline HPXMLtoOpenStudio/resources/hvac.rb:3340]: 👍

File: HPXMLtoOpenStudio/resources/hvac_sizing.rb [hunk 2]
```diff
@@ -4972,22 +4988,23 @@ def initialize
...
-                :Heat_Capacity,       # [Double] Nominal heating capacity (Btu/hr)
-                :Heat_Capacity_Supp,  # [Double] Nominal heating capacity for the HP backup (Btu/hr)
-                :Cool_Airflow,        # [Double] Cooling airflow rate (cfm)
-                :Heat_Airflow,        # [Double] Heating airflow rate (cfm)
-                :GSHP_Loop_Flow,      # [Double] Ground-source heat pump water flow rate through the geothermal loop (gal/min)
-                :GSHP_Bore_Holes,     # [Integer] Ground-source heat pump number of boreholes (#)
-                :GSHP_Bore_Depth,     # [Double] Ground-source heat pump depth of each borehole (ft)
-                :GSHP_G_Functions,    # [Array<Array<Double>, Array<Double>>] Ground-source heat pump G-functions
-                :GSHP_Bore_Config)    # [String] Ground-source heat pump borefield configuration (HPXML::GeothermalLoopBorefieldConfigurationXXX)
+  attr_accessor(:Cool_Load_Sens,      # Total sensible cooling load (Btu/hr)
```
Comment (shorowit) [inline HPXMLtoOpenStudio/resources/hvac_sizing.rb:4991]: When you resolved conflicts, you reverted a bunch of changes here and below that we want. Can you fix this?

File: HPXMLtoOpenStudio/resources/hvac_sizing.rb [hunk 3]
```diff
@@ -4972,22 +4988,23 @@ def initialize
...
+                :Cool_Load_Lat,       # Total latent cooling load (Btu/hr)
+                :Cool_Load_Tot,       # Total (sensible + latent) cooling load (Btu/hr)
+                :Heat_Load,           # Total heating sensible load (Btu/hr)
+                :Heat_Load_Supp,      # Total heating sensible load for the HP backup (Btu/hr)
+                :Cool_Capacity,       # Nominal total cooling capacity (Btu/hr)
+                :Cool_Capacity_Sens,  # Nominal sensible cooling capacity (Btu/hr)
+                :Heat_Capacity,       # Nominal heating capacity (Btu/hr)
+                :Heat_Capacity_Supp,  # Nominal heating capacity for the HP backup (Btu/hr)
+                :Cool_Airflow,        # Cooling airflow rate (cfm)
+                :Design_SHR,          # Cooling design SHR (dimensionless)
```
Comment (shorowit) [inline HPXMLtoOpenStudio/resources/hvac_sizing.rb:5001]: No need to add this property to the object. Just use a local variable in the `apply_hvac_equipment_adjustments` method (e.g., `design_shr` instead of `hvac_sizings.Design_SHR`).

File: HPXMLtoOpenStudio/resources/hvac_sizing.rb [hunk 4]
```diff
@@ -3611,24 +3644,7 @@ def self.calc_duct_leakages_cfm25(distribution_system, system_cfm)
...
-  # This method is being deleted in https://github.com/NREL/OpenStudio-HPXML/pull/1707
-  def self.get_shr_biquadratic
-    # Based on EnergyPlus's model for calculating SHR at off-rated conditions. This curve fit
-    # avoids the iterations in the actual model. It does not account for altitude or variations
-    # in the SHRRated. It is a function of ODB (MJ design temp) and CFM/Ton (from MJ)
-    return [1.08464364, 0.002096954, 0, -0.005766327, 0, -0.000011147]
-  end
-
-  # Determines the nominal speed (of a multi/variable-speed system).
+  # Determines the speed (of a multi/variable-speed system) that is running during the design conditions.
```
Comment (shorowit) [inline HPXMLtoOpenStudio/resources/hvac_sizing.rb:3647]: Looks like you reverted a change here too. Please fix.

**Review by shorowit — COMMENTED** [review]
I'll probably clean up the code a bit before merging to remove unnecessary comments, but overall it looks pretty reasonable from a programming perspective. Just a couple comments. Relying on @jonwinkler for the engineering review.

File: HPXMLtoOpenStudio/resources/hvac_sizing.rb (hunk 3 above)
Comment (Sashadf1) [reply HPXMLtoOpenStudio/resources/hvac_sizing.rb:5001]: @shorowit ok. will update the variable name.

Comment (Sashadf1) [reply HPXMLtoOpenStudio/resources/hvac_sizing.rb:5001]: refactor done, changes committed.

File: HPXMLtoOpenStudio/resources/hvac_sizing.rb (hunk 4 above)
Comment (Sashadf1) [reply HPXMLtoOpenStudio/resources/hvac_sizing.rb:3647]: @shorowit I made some local corrections to the comments above self.get_nominal_speed, committed them, and performed a git diff between the remote branch adp_bf_method_design_shr and remote master. In VS code terminal this is 

git diff adp_bf_method_design_shr origin/master  

There is no diff for the get_nominal_speed method now. What is in adp_bf branch matches master exactly.

File: HPXMLtoOpenStudio/resources/hvac_sizing.rb (hunk 2 above)
Comment (Sashadf1) [reply HPXMLtoOpenStudio/resources/hvac_sizing.rb:4991]: I see the diffs, they were just in the comments around the HVACSizingValues class. I resolved them to match master and committed.

File: HPXMLtoOpenStudio/resources/hvac_sizing.rb [hunk 5]
```diff
@@ -2137,101 +2145,126 @@ def self.apply_hvac_equipment_adjustments(mj, runner, hvac_sizings, weather, hva
...
 
-      # Calculate the air flow rate required for design conditions
+      rated_barometric_pressure_psi = UnitConversions.convert(1, 'atm', 'psi') # assume rated pressure is at sea level
+
+      hr_indoor_cooling_rated = Psychrometrics.w_fT_Twb_P(HVAC::AirSourceCoolRatedIDB, HVAC::AirSourceCoolRatedIWB, rated_barometric_pressure_psi) # [lbm/lbm]
+
+      # calculate A_o using rated conditions
+      # Don't use coil ao factor in hvac.rb to calculate design SHR. That ao has specific use case of finding rated SHRs for variable speed equipment.
+      a_o_rated = Psychrometrics.CoilAoFactor(HVAC::AirSourceCoolRatedIDB, rated_barometric_pressure_psi, UnitConversions.convert(cool_cap_rated, 'btu/hr', 'kbtu/hr'), cool_cfm_rated, hvac_cooling_shr, hr_indoor_cooling_rated)
+      cool_cap_design = hvac_sizings.Cool_Load_Tot
```
Comment (jonwinkler) [inline HPXMLtoOpenStudio/resources/hvac_sizing.rb:2162]: @Sashadf1 Maybe add some comments above this line indicating this is an initial assumption prior to attempting to balance sensible and latent design capacity by applying the over size limits.

File: HPXMLtoOpenStudio/resources/hvac_sizing.rb [hunk 6]
```diff
@@ -2137,101 +2145,126 @@ def self.apply_hvac_equipment_adjustments(mj, runner, hvac_sizings, weather, hva
...
-                                  (a_sens + c_sens * entering_temp) - 1.0)
+        design_shr = Psychrometrics.CalculateSHR(mj.cool_setpoint, mj.p_psi, UnitConversions.convert(cool_cap_design, 'btu/hr', 'kbtu/hr'), hvac_sizings.Cool_Airflow, a_o_rated, hr_indoor_cooling_design)
+        # Calculate the coil SHR at the given incoming air state, CFM, total capacity, and coil Ao factor
+        # CFM changes in the iteration based on current value of design_shr
 
-        # Ensure equipment is not being undersized
-        cool_load_sens_cap_design = [cool_load_sens_cap_design, undersize_limit * hvac_sizings.Cool_Load_Sens].max
+        # calculate sensible/latent split at design conditions
+        cool_sens_cap_design = cool_cap_design * design_shr
+        cool_lat_cap_design = [hvac_sizings.Cool_Load_Tot - cool_sens_cap_design, 1.0].max
```
Comment (jonwinkler) [inline HPXMLtoOpenStudio/resources/hvac_sizing.rb:2191]: @Sashadf1, Should this be:

`cool_lat_cap_design = [cool_cap_design - cool_sens_cap_design, 1.0].max`

I think `cool_cap_design` may change as the loop iterates, right?

File: HPXMLtoOpenStudio/resources/hvac_sizing.rb (hunk 5 above)
Comment (Sashadf1) [reply HPXMLtoOpenStudio/resources/hvac_sizing.rb:2124]: ok, will do.

Can you clarify the line number, and which initial assumption?

File: HPXMLtoOpenStudio/resources/hvac_sizing.rb (hunk 6 above)
Comment (Sashadf1) [reply HPXMLtoOpenStudio/resources/hvac_sizing.rb:2153]: yes, cool_cap_design may change as the loop iterates.

File: HPXMLtoOpenStudio/resources/hvac_sizing.rb (hunk 5 above)
Comment (jonwinkler) [reply HPXMLtoOpenStudio/resources/hvac_sizing.rb:2126]: I was thinking the line `cool_cap_design = hvac_sizings.Cool_Load_Tot` could use an explanation. I know it's obvious based the code that comes next but I thought additional comments would be worth it.

File: HPXMLtoOpenStudio/resources/hvac_sizing.rb [hunk 7]
```diff
@@ -2099,95 +2107,122 @@ def self.apply_hvac_equipment_adjustments(mj, runner, hvac_sizings, weather, hva
...
+          # Limit total capacity to oversize limit
+          cool_cap_design = [cool_cap_design, oversize_limit * hvac_sizings.Cool_Load_Tot].min
 
-        cool_cap_design = (sens_cap_design / (hvac_cooling_shr / total_cap_curve_value) - \
-                                           (b_sens * UnitConversions.convert(hvac_sizings.Cool_Airflow, 'ton', 'Btu/hr') + \
-                                           d_sens * UnitConversions.convert(hvac_sizings.Cool_Airflow, 'ton', 'Btu/hr') * entering_temp)) / \
-                          (a_sens + c_sens * entering_temp)
+          # Determine rated capacities
+          hvac_sizings.Cool_Capacity = cool_cap_design / total_cap_curve_value
+          hvac_sizings.Cool_Capacity_Sens = hvac_sizings.Cool_Capacity * hvac_cooling_shr
```
Comment (jonwinkler) [inline HPXMLtoOpenStudio/resources/hvac_sizing.rb:2187]: Consider this optional, but renaming `hvac_cooling_shr` to something like `hvac_cooling_shar_rated` might be nice for clarity.

Comment (Sashadf1) [reply HPXMLtoOpenStudio/resources/hvac_sizing.rb]: done, clarity rename to hvac_cooling_shr_rated

**Comment by Sashadf1** [comment]
@shorowit to resolve merge conflicts and coordinate w/ residential team to test w/ diverse buildstock run

**Review by shorowit — APPROVED** [review]
We have pounded the tires on this enough. Thanks for the good work here, @Sashadf1. I'll merge this in once the CI comes back happy.
